pkg/controller/clustersync/clustersync_controller.go

```diff
@@ -517,10 +520,23 @@ func (r *ReconcileClusterSync) applySyncSets(
 				startTime = cond.LastTransitionTime.Time
 			}
 			applyTime := now.Sub(startTime).Seconds()
+
 			if syncSet.AsMetaObject().GetNamespace() == "" {
-				logger.WithField("applyTime", applyTime).Debug("observed first successful apply of SelectorSyncSet for cluster")
-				metricTimeToApplySelectorSyncSet.WithLabelValues(syncSet.AsMetaObject().GetName()).Observe(applyTime)
+				if reportSelectorSyncSetMetrics {
+					// Report SelectorSyncSet metric *only if* we have not yet reached our "everything has applied successfully for the first time"
+					// state. This is to handle situations where a clusters labels change long after it was installed, resulting
+					// in a new SelectorSyncSet matching, and a metric showing days/weeks/months time to first apply. In this scenario
+					// we have no idea when the label was added, and thus it is not currently possible to report the time from
+					// label added to SyncSet successfully applied.
+					logger.WithField("applyTime", applyTime).Debug("observed first successful apply of SelectorSyncSet for cluster")
+					metricTimeToApplySelectorSyncSet.WithLabelValues(syncSet.AsMetaObject().GetName()).Observe(applyTime)
+				} else {
+					log.WithField("SelectorSyncSet", syncSet.AsMetaObject().GetName()).
```

Comment:
This logger field is already set [here](https://github.com/openshift/hive/blob/4bd6d19e745bcdea3a59bdf85756a17f6c53ac89/pkg/controller/clustersync/clustersync_controller.go#L435)

Reply:
Also just noticed I used log instead of logger. Oops.